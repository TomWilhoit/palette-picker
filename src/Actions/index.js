export const addProjects = (projects) => ({
  type: 'ADD_PROJECTS',
  projects
})

export const addPalettes = (palettes) => ({
  type: 'ADD_PALETTES',
  palettes
})

export const addCurrentProject = (project) => ({
  type: 'ADD_CURRENT_PROJECT',
  project
})