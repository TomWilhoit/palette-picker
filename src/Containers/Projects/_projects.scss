.projects-container {
  @include container;
  background-color: $light-background;
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 100%;
}

.projects-head {
  border-bottom: 3px solid $border;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 50%;
}

.projects-container h3 {
  padding: 0;
}

.projects-list {
  background-color: #e6e3df;
  height: 50%;
  overflow-y: scroll;
  overflow-x: hidden;
}

.projects-list::-webkit-scrollbar {
  -webkit-box-shadow: inset 0 0 5px rgba(0,0,0,0.4);
  background-color: #F5F5F5;
  width: 14px;
}

.projects-list::-webkit-scrollbar-thumb {
	border-radius: 8px;
	-webkit-box-shadow: inset 0 0 5px rgba(0,0,0,.4);
	background-color: rgb(141, 175, 207);
}

.projects-list::-webkit-scrollbar-track {
	-webkit-box-shadow: inset 0 0 3px rgba(0,0,0,0.3);
	border-radius: 10px;
	background-color: #F5F5F5;
}

.new-project {
  height: 30%;
  margin: 0;
}