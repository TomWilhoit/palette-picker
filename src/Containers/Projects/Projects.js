import React, { Component } from "react";
import { connect } from "react-redux";
import { addProjects } from "../../Actions/index";
import Project from "../Project/Project";
import NewProject from "../NewProject/NewProject";
import PropTypes from "prop-types";

export class Projects extends Component {

  renderProjects = () => {
    if (this.props.projects.length) {
      return this.props.projects.map((project, index) => (
        <Project name={project.name} key={index} id={project.id} />
      ));
    }
  } 

  render() {
    const displayProjects = this.renderProjects()
      return (
        <div className="projects-container">
          <NewProject />
          {displayProjects}
        </div>
      );
  }
}

Projects.propTypes = {
  projects: PropTypes.array,
  palettes: PropTypes.array,
  currentProject: PropTypes.number
};

export const mapStateToProps = state => ({
  projects: state.projects,
  palettes: state.palettes,
  currentProject: state.currentProject
});

export const mapDispatchToProps = dispatch => ({
  addProjects: project => dispatch(addProjects(project))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Projects);
