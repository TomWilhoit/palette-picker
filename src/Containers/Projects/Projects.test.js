import React from "react";
import { Projects } from "./Projects";
import { mapStateToProps } from "./Projects";
import { mapDispatchToProps } from "./Projects";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { addProjects } from "../../Actions";



describe("Projects", () => {
  let wrapper;
  let mockProjects = [{ name: "Tom"},{name: "Mason"}];
  beforeEach(() => {
    wrapper = shallow(<Projects projects={mockProjects} />);
  });

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should mapStateToProps", () => {
    const mockState = {
      currentProject: 4,
      palettes: [{ name: "Mason", projectId: 4 }],
      projects: [{name: "Tommy"}]
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(mockState);
  });

  it("should map dispatch to props", () => {
    const mockProject= {name: "Tommy"}
    const mockDispatch = jest.fn();
    const actionToDispatch = addProjects(mockProject);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.addProjects(mockProject);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});