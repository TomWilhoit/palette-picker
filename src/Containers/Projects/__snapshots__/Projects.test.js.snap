// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should match the snapshot with all data passed in 1`] = `
<div
  className="projects-container"
>
  <ConnectFunction />
  <ConnectFunction
    id={2}
    key="0"
    name="Tom"
  />
  <ConnectFunction
    id={3}
    key="1"
    name="Mason"
  />
</div>
`;
